### ISSUE TITLE
SP metadata requested attribute elements without NameFormat cause unhandled exception

### ISSUE THREAD

**Issue by skoranda** [IdentityPython/pysaml2#466]
If the SP has metadata that includes a \<RequestedAttribute\> without a NameFormat attribute then a KeyError is thrown from  method _match_attr_name() in assertion.py. The reason is that the attr dictionary object being passed to get_local_name() will not in that case have the key "name_format".

**Comment by c00kiemon5ter** [comment]
As @skoranda mentioned, this is [the commit](https://github.com/rohe/pysaml2/commit/5bcb6ac24a15c40a74cfd0756a5dfc045ad155cf) that introduced the regression. 

If `attr["name_format"]` not set, can we assume that it is `urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified` (`NAME_FORMAT_UNSPECIFIED`).

**Comment by c00kiemon5ter** [comment]
if the above holds, then we can fallback to that

```python
from saml2.saml import NAME_FORMAT_UNSPECIFIED

def _match_attr_name(attr, ava):
  local_name = get_local_name(acs, attr["name"], attr.get("name_format", NAME_FORMAT_UNSPECIFIED))
  ...
```

else, lets set the vars outside the function call

```python
def _match_attr_name(attr, ava):
  attr_name = attr.get("name")
  attr_name_format = attr.get("name_format")
  if attr_name_format:
    local_name = get_local_name(acs, attr_name, attr_name_format)
  else:
    local_name = None
  ...
```

**Comment by c00kiemon5ter** [comment]
This has been fixed by f6a47025f034fc030a6cbb8b7ec901399b58118d
